### PR TITLE
[refactor] #3752: Replace `MockValidator` with `Initial`

### LINKED ISSUE #3752: [refactor]: Replace `MockValidator` with `Default`
Maybe we can design so that when empty wsv is created it's already have minimal validator that only execute instructions (and `Upgrade` instruction in genesis will replace it)?

By doing that we can keep design more uniform.

_Originally posted by @Erigara in https://github.com/hyperledger/iroha/pull/3723#discussion_r1272997794_

### PR DESCRIPTION
**Pull request by Arjentix**
## Description

- Removed `MockValidator` and its usage with `#[cfg(test)]`
- `Validator` is now an enum with `Initial` variant set by default
- This `Initial` variant will perform no permission checking, just operation execution

### Linked issue

- Closes #3752 

### Benefits

No divergence between real and test code

### Question

Should we allow genesis without Validator then?

### Checklist

- [x] I've read `CONTRIBUTING.md`
- [x] I've used the standard signed-off commit format (or will squash just before merging)
- [x] All applicable CI checks pass (or I promised to make them pass later)
- [ ] (optional) I've written unit tests for the code changes
- [ ] I replied to all comments after code review, marking all implemented changes with thumbs up

### CODE REVIEW

File: core/src/wsv.rs [hunk 1]
```diff
@@ -976,9 +952,7 @@ impl WorldStateView {
     /// Panics if validator is not initialized.
     /// Possible only before applying genesis.
```
Comment (Erigara) [inline core/src/wsv.rs:953]: I belive panic docs in no longer true, right?

File: core/src/smartcontracts/isi/world.rs [hunk 2]
```diff
@@ -218,32 +218,21 @@ pub mod isi {
...
-                        .migrate(wsv, authority)
-                        .map(|migration_result| migration_result.map(|_: ()| new_validator))
-                })
-                .map_err(|error| {
-                    InvalidParameterError::Wasm(format!("{:?}", eyre::Report::from(error)))
-                })?
+
+            // Cloning validator to avoid multiple mutable borrows.
+            // Also it's a cheap operation.
+            let mut cloned_validator = wsv.validator().clone();
```
Comment (Erigara) [inline core/src/smartcontracts/isi/world.rs:225]: Maybe `upgraded_validator`?

**Review by Erigara — APPROVED** [review]
Great this change allowed to throw away mocking code.

**Review by mversic — DISMISSED** [review]

File: core/src/validator.rs [hunk 3]
```diff
@@ -77,18 +77,33 @@ impl Validator {
...
-            transaction,
-        )?
+
+        match self {
+            Self::Initial => {
+                let (_authority, Executable::Instructions(instructions)) = transaction.into() else {
+                    return Ok(());
+                };
+                for isi in instructions {
+                    self.validate_instruction(wsv, authority, isi)?;
```
Comment (mversic) [inline core/src/validator.rs:88]:
```suggestion
                    isi.execute(authority, wsv).map_err(Into::into)?;
```

I would rather have it like this

**Review by mversic — APPROVED** [review]

**Pull request merged**